cinema/tests/test_movie_api.py

```diff
@@ -157,3 +156,145 @@ def test_image_url_is_shown_on_movie_session_detail(self):
         res = self.client.get(MOVIE_SESSION_URL)
 
         self.assertIn("movie_image", res.data[0].keys())
+
+
+class UnAuthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        result = self.client.get(MOVIE_URL)
+        self.assertEqual(result.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            "user@myproject.com", "password"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_list_movies(self):
+        sample_movie()
+        genre = sample_genre()
+        genre2 = sample_genre(name="Action")
+        movie_with_genres = sample_movie()
+        movie_with_genres.genres.add(genre2, genre)
+        actor = sample_actor()
+        movie_with_actor = sample_movie()
+        movie_with_actor.actors.add(actor)
+        result = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+        self.assertEqual(result.status_code, status.HTTP_200_OK)
+        self.assertEqual(result.data, serializer.data)
```

Comment:
add empty lines between genres and movies or try to place more logic in one line. It will make more readable

Reply:
fix in all tests